odoo/models.py

```diff
@@ -3722,6 +3716,11 @@ def _write(self, vals):
         # determine records that require updating parent_path
         parent_records = self._parent_store_update_prepare(vals)
 
+        if self._log_access:
+            vals = dict(vals)
+            vals.setdefault('write_uid', self.env.uid)
+            vals.setdefault('write_date', self.env.cr.now())
```

Comment:
Is this in case `_write` gets called directly? Do we really care?

Reply:
The fields `write_uid` and `write_date` are already assigned by `write()`, and will be part of `vals` when flushing the pending updates.  However, those fields are not assigned when a computed field is assigned; in that case, this code ensures that both fields are updated anyway.